### DISCUSSION TITLE
Need to increase SPI read-write speed

### DISCUSSION THREAD

**Discussion by arieslaptop** [nofeletru/UsbAsp-flash#65] [Ideas]
Thanks for making a wonderful application 
Need to increase SPI read-write speed  (Tested with CH341A & CJMCU FT232H) (16MB-W25Q128FV)

                     
### **FT232H                   7Sec (Read)              03M35Sec(Write)
**CH341A              2M30Sec (Read)              8M11Sec (Write)****

### **FT232H RESULT**
AsProgrammer v2.0.3a
https://github.com/nofeletru/UsbAsp-flash
asprogrammer@inbox.ru
Current programmer: FT232H
13:47:24
Reading memory...
Done
Execution time: 0:00:07
CRC32 = 0xDEFD69BC
Current programmer: FT232H
ID(9F): EF4018
ID(90): EF17
ID(AB): 17
ID(15): 6060
Current programmer: FT232H
13:48:12
Reading memory...
Done
Execution time: 0:00:06
CRC32 = 0xDEFD69BC
Current programmer: FT232H
Current programmer: FT232H
Old SREG: 00000000(0x00), 00000010(0x02)
New SREG: 00000000(0x00), 00000010(0x02)
Current programmer: FT232H
13:49:31
Erasing memory...
The process can take more than a minute on large flash!
Done
Execution time: 0:00:45
Current programmer: FT232H
13:50:17
Programming memory (with verification)...
Done
Execution time: 0:03:35

### **CH341A RESULT** 
AsProgrammer v2.0.3a
https://github.com/nofeletru/UsbAsp-flash
asprogrammer@inbox.ru
Current programmer: CH341
ID(9F): EF4018
ID(90): EF17
ID(AB): 17
ID(15): 6060
Current programmer: CH341
13:16:05
Reading memory...
Done
Execution time: 0:02:30
CRC32 = 0xDEFD69BC
Current programmer: CH341
Old SREG: 00000000(0x00), 00000010(0x02)
New SREG: 00000000(0x00), 00000010(0x02)
Current programmer: CH341
13:18:42
Erasing memory...
The process can take more than a minute on large flash!
Done
Execution time: 0:00:41
Current programmer: CH341
13:19:23
Programming memory (with verification)...
Done
Execution time: 0:08:11

**Comment by nofeletru** [comment]
I don't think I can squeeze out more speed.
